context-config-store/store/context_config_store_impl.go

```
@@ -0,0 +1,139 @@
+	}
+
+	currentContextUuid := contextConfig.GetCurrentContext()
+	var knownContexts []string
+	for _, kurtosisContext := range contextConfig.GetContexts() {
+		knownContexts = append(knownContexts, kurtosisContext.GetUuid().GetValue())
+		if kurtosisContext.GetUuid().GetValue() == currentContextUuid.GetValue() {
+			return kurtosisContext, nil
+		}
+	}
+	return nil, stacktrace.NewError("Unable to find current context info in context config file. "+
+		"Current context is set to '%s' but known contexts are: '%s'",
+		currentContextUuid.GetValue(),
+		strings.Join(knownContexts, ", "))
+}
+
+func (store *contextConfigStoreImpl) SwitchContext(contextUuid *api.ContextUuid) error {
+	store.Lock()
+	defer store.Unlock()
+
+	contextConfig, err := store.storage.LoadContextConfig()
+	if err != nil {
+		return stacktrace.Propagate(err, "Unable to load current context config")
+	}
+
+	var doesExist bool
+	for _, kurtosisContext := range contextConfig.GetContexts() {
+		doesExist = doesExist || kurtosisContext.GetUuid().GetValue() == contextUuid.Value
+	}
+	if !doesExist {
+		return stacktrace.NewError("Context with UUID '%s' does not exist yet.", contextUuid.GetValue())
+	}
+
+	newContextConfig := api.NewKurtosisContextConfig(contextUuid, contextConfig.GetContexts()...)
+	if err = store.storage.PersistContextConfig(newContextConfig); err != nil {
+		return stacktrace.Propagate(err, "Unable to persist new current context to store")
+	}
+	return nil
+}
+
+func (store *contextConfigStoreImpl) AddNewContext(contextToAdd *api.KurtosisContext) error {
+	store.Lock()
+	defer store.Unlock()
+
+	contextConfig, err := store.storage.LoadContextConfig()
+	if err != nil {
+		return stacktrace.Propagate(err, "Unable to load current context config")
+	}
+
+	newContextUuid := contextToAdd.GetUuid()
+	for _, kurtosisContext := range contextConfig.GetContexts() {
+		if kurtosisContext.GetUuid().GetValue() == newContextUuid.GetValue() {
+			return nil // doing nothing as context already exists
```

interesting that we choose to return nil instead of erroring (fail loudly) like a dictionary. can this have side effects? will the person assume that the context they are storing got stored while the previous one stays persisted?

---

Yes you're right, I think it's probably safer to throw loudly here. If users want to add a context that already exists, we can suggest them to remove it first and then adding it again. I'll throw here with a clear message, good suggestion!